Trisha-tech/OnlineBookSales
# Added the backend part for Google Sign IN

This PR does the following:
Adds Google authentication functionality to the application.
Updates the server configuration to handle OAuth authentication.
Fixes potential CORS issues to allow frontend requests.

Related Issues
Fixes #274 

Changes
Implemented Google OAuth 2.0 authentication using the passport-google-oauth20 strategy.
Configured environment variables to securely store Google Client ID and Client Secret.
Added a new endpoint for handling Google authentication callbacks.

Testing Instructions
Detailed instructions on how to test the changes:

Pull this branch.
Run npm install to install dependencies.
Ensure that your .env file includes valid GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET.
Start the server using npm start.
Navigate to http://localhost:5000/auth/google to initiate Google authentication.
Verify that users can log in using their Google accounts and that the user profile is correctly handled in the application.


Additional Context
Any additional context or information that reviewers should be aware of:

This PR is based on the integration of Google authentication to improve user experience and streamline the login process.
This is the backend part, the google sign in button still need to be designed at the frontend and redirected properly.
I request the publisher to please generate Google OAuth client id and secret to proceed with the linkage in the website.

Checklist
 [x] I have performed a self-review of my code
 [x] I have commented on my code, particularly in hard-to-understand areas
 [x] My changes generate no new warnings
 [x] I am working on this issue under GSSOC
